pkg/sbom/core/bom.go

```diff
@@ -203,7 +203,6 @@ type Vulnerability struct {
 
 type Options struct {
 	GenerateBOMRef bool // Generate BOMRef for CycloneDX
-	Parents        bool // Hold parent maps
```

Comment:
Currently, our SBOM scanning process involves several steps:

1. [Decoding the SBOM to create a `core.SBOM`](https://github.com/aquasecurity/trivy/blob/0e286f074f5bc584b05132f5b4652858da35df17/pkg/sbom/cyclonedx/unmarshal.go#L25)
2. [Converting this `core.SBOM` to a `types.SBOM`](https://github.com/aquasecurity/trivy/blob/0e286f074f5bc584b05132f5b4652858da35df17/pkg/sbom/io/decode.go#L50C59-L50C63)
3. For certain operations (e.g., VEX application), [re-encoding the `types.Report` back to a `core.SBOM`](https://github.com/aquasecurity/trivy/blob/0e286f074f5bc584b05132f5b4652858da35df17/pkg/result/filter.go#L91)

We've identified that this process, particularly the decoding and re-encoding between `core.SBOM` and `types.Report`, is not as efficient as it could be. Ideally, we should reuse the decoded SBOM instead of re-encoding it.

One key challenge I've encountered is that during the initial SBOM decoding, we can't always determine whether "parents" will be needed later in the process (applying VEX or not). Given this uncertainty, I decided to always generate the parents.

I attempted to implement the reuse of the decoded SBOM, but found that it required more extensive modifications to our codebase than I initially anticipated. As a result, I've reverted these changes for now to maintain code stability and manageability.

Looking ahead, I still plan to remove the parent field in question to facilitate this optimization in the future. However, I'm keeping the option open to reintroduce this field if needed in the short term.

Reverted in https://github.com/aquasecurity/trivy/pull/7206/commits/9881b0fd3b43455f1c2cc22458aa95eb0420817c